app/src/main/java/org/eyeseetea/malariacare/data/authentication/AuthenticationManager.java

```diff
@@ -34,7 +34,7 @@ public AuthenticationManager(Context context) {
     @Override
     public void login(final Credentials credentials,
             final IAuthenticationManager.Callback<UserAccount> callback) {
-        if (credentials.isDemoCredentials()) {
+        if (credentials == null || credentials.isDemoCredentials()) {
```

Comment:
if credentials is null on login we should throw IllegalArgumentException because null credentials and demo credentials is not the same case